pylint/checkers/base.py

```diff
@@ -1218,6 +1303,32 @@ def open(self):
             for name_type in group.split(':'):
                 self._name_group[name_type] = 'group_%s' % (group,)
 
+        regexps, hints = self._create_naming_rules()
+        self._name_regexps = regexps
+        self._name_hints = hints
+
+    def _create_naming_rules(self):
+        regexps = {}
+        hints = {}
+
+        for name_type in KNOWN_NODE_TYPES:
+            naming_style_option_name = "%s_naming_style" % (name_type,)
+            naming_style_name = getattr(self.config, naming_style_option_name)
+
+            regexps[name_type] = NAMING_STYLES[naming_style_name].get_regex(name_type)
+
+            custom_regex_setting_name = "%s_rgx" % (name_type, )
```

Comment:
There is a problem with picking the rgx if it exists, because currently we will always pick regexes, due to already having the ``-rgx`` hints pre-generated in config files. One idea would be to rename the options to something else, so for ``attr-rgx`` for instance, we should have ``attr-pattern`` or something along these lines. Could you think of another approach of how we should handle this?